types/plugin.d.ts

```diff
@@ -128,14 +128,14 @@ declare namespace fastifyCookie {
   type HookType = 'onRequest' | 'preParsing' | 'preValidation' | 'preHandler'  | 'preSerialization';
 
   export interface FastifyCookieOptions {
-    secret?: string | string[] | Signer;
+    secret?: string | string[] | Buffer | Buffer[] | Signer;
     hook?: HookType | false;
     parseOptions?: fastifyCookie.CookieSerializeOptions;
   }
 
-  export type Sign = (value: string, secret: string, algorithm?: string) => string;
-  export type Unsign = (input: string, secret: string, algorithm?: string) => UnsignResult;
-  export type SignerFactory = (secrets: string | Array<string>, algorithm?: string) => SignerBase;
+  export type Sign = (value: string, secret: string | Buffer, algorithm?: string) => string;
+  export type Unsign = (input: string, secret: string | Buffer, algorithm?: string) => UnsignResult;
+  export type SignerFactory = (secrets: string | Array<string> | Buffer | Array<Buffer>, algorithm?: string) => SignerBase;
```

Comment:
I guess we can use only one notation through out the types: 

```suggestion
  export type SignerFactory = (secrets: string | string[] | Buffer | Buffer[], algorithm?: string) => SignerBase;
```